### PR TITLE
feat. add repo public for hubauth svc.

### PR DESCRIPTION
**Pull request by lingdie**

### CODE REVIEW

File: service/hub/server/config.go [hunk 1]
```diff
@@ -82,6 +102,8 @@ func LoadConfig(fileName string) (*Config, error) {
 	if err = validate(c); err != nil {
 		return nil, fmt.Errorf("invalid config: %s", err)
 	}
+	// store client globally
+	k8sClient, _ = kubernetes.NewKubernetesClient(c.Server.Kubeconfig, "")
```
Comment (lingdie) [inline service/hub/server/config.go:106]: split cheaker and var initer.

File: service/hub/server/config.go [hunk 2]
```diff
@@ -8,10 +8,23 @@ import (
...
+	"github.com/cesanta/glog"
 	"github.com/docker/libtrust"
+	"github.com/labring/sealos/pkg/client-go/kubernetes"
 	yaml "gopkg.in/yaml.v2"
+	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+var k8sClient kubernetes.Client
+
+func init() {
```
Comment (lingdie) [inline service/hub/server/config.go:20]: why create var when first use anonymous

File: service/hub/server/config.go (hunk 1 above)
Comment (zzjin) [reply service/hub/server/config.go:106]: Why reassign global k8sClient to new user spec kubeconfig?

Comment (lingdie) [reply service/hub/server/config.go]: deleted.

File: service/hub/server/config.go [hunk 3]
```diff
@@ -98,5 +102,10 @@ func LoadConfig(fileName string) (*Config, error) {
 	if !tokenConfigured {
 		return nil, fmt.Errorf("failed to load token cert and key: none provided")
 	}
+	// setup k8sClient by using controller-runtime
+	k8sClient, err = kubernetes.NewKubernetesClientByConfig(ctrl.GetConfigOrDie())
```
Comment (zzjin) [inline service/hub/server/config.go:106]: init here, but k8sClient never called?

**Review by zzjin — APPROVED** [review]

**Pull request merged**